Source/Timeline.swift

```diff
@@ -0,0 +1,133 @@
+//
+//  Timeline.swift
+//  VEAudioKit
+//
+//  Created by Pablo Balduz on 19/08/2019.
+//
+
+import AVFoundation
+
+// MARK: - TimelineDelegate
+
+/// The `TimelineDelegate` provides an interface to respond to `Timeline` events as well as to configure it.
+protocol TimelineDelegate: class {
+    
+    var length: TimeInterval { get }
+    
+    /// A `TimeInterval` to control the time updates triggering rate
+    var timeUpdateRate: TimeInterval { get }
+    
+    /// Triggered when a time update happens according to `timeUpdateRate` property value
+    ///
+    /// - Parameters:
+    ///   - time: The time elapsed (expressed in seconds) since the `Timeline` instance was started
+    func currentTimeDidUpdate(time: Double)
+    
+    func timelineDidReachEnd()
+}
+
+extension TimelineDelegate {
+    var timeUpdateRate: TimeInterval { return 0.1 }
+}
+
+// MARK: - Timeline
+
+/// Custom timeline designed to work while playing audio files.
+/// It provides the ability to track the current position of the audio expressed in seconds.
+/// It also provides a `TimelineDelegate` that enables responding to time updates.
+class Timeline {
+    
+    // MARK: - Public properties
+    
+    /// A `Double` representing the current second in the timeline
+    var currentTime: Double {
+        guard let length = delegate?.length else { return 0 }
+        let time = lastPauseTime + currentElapsedTime
+        return max(min(length, time), 0)
```

Comment:
I'm starting to think about taking this into a sepparate function, extension of Double or Float and accepting a range as a param.
Example:
`time.bound(by: [0...length])`